### PR TITLE
#100 Promotions Rework - No longer preliminary, since no one's merged it.....

### PR DESCRIPTION
**Pull request by JakenVeina**
New features and integrations, in preparation for the full promotions rework.

- Role Tracking & Designation Features
- Channel Designation Feature Cleanup (for consistency with Role Designation)
- Also a migration bugfix.
- Moderation Mute Role system integrated into Role Designation Feature

### CODE REVIEW

File: Modix.Data/Models/Promotions/PromotionCommentCreationData.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+{
+    /// <summary>
+    /// Describes an operation to create a <see cref="PromotionCommentEntity"/>.
+    /// </summary>
+    public class PromotionCommentCreationData
+    {
+        /// <summary>
+        /// See <see cref="PromotionCommentEntity.CampaignId"/>.
+        /// </summary>
+        public long CampaingId { get; set; }
```
Comment (jmazouri) [inline Modix.Data/Models/Promotions/PromotionCommentCreationData.cs]: Small typo here - should be "CampaignId"

File: Modix.Data/Repositories/PromotionCampaignRepository.cs [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            {
+                GuildId = entity.GuildId,
+                Type = PromotionActionType.CampaignClosed,
+                Created = DateTimeOffset.Now,
+                CreatedById = (long)closedById,
+                CampaignId = entity.Id
+            };
+            await ModixContext.SaveChangesAsync();
+
+            await RaisePromotionActionCreatedAsync(entity.CreateAction);
```
Comment (jmazouri) [inline Modix.Data/Repositories/PromotionCampaignRepository.cs]: This should be "entity.CloseAction", no?

Comment (JakenVeina) [reply Modix.Data/Repositories/PromotionCampaignRepository.cs]: Ayup.

Comment (JakenVeina) [reply Modix.Data/Repositories/PromotionCampaignRepository.cs]: Ayup

File: Modix.Data/Models/Promotions/PromotionCommentCreationData.cs (hunk 1 above)
Comment (JakenVeina) [reply Modix.Data/Models/Promotions/PromotionCommentCreationData.cs]: Ayup

**Pull request merged**